core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -155,7 +155,13 @@ private[spark] abstract class Task[T](
    */
   def collectAccumulatorUpdates(taskFailed: Boolean = false): Seq[AccumulatorV2[_, _]] = {
     if (context != null) {
-      context.taskMetrics.accumulators().filter { a => !taskFailed || a.countFailedValues }
+      context.taskMetrics.internalAccums.filter { a =>
+        // RESULT_SIZE accumulator is always zero at executor, we need to send it back as its
+        // value will be updated at driver side.
+        !a.isZero || a.name == Some(InternalAccumulator.RESULT_SIZE)
+      // zero value external accumulators may still be useful, e.g. SQLMetrics, we should not filter
```

Comment:
If there is no spilling, we could say the size of spilling is not define (null for unknown). We also have total value, we could know that how many task had spilled. Right now, we can't know how many tasks had spilled, actually it's worse.

I don't think it's wrong.